src/Agda2Hs/Compile/Type.hs

```diff
@@ -184,14 +186,21 @@ compileTeleBinds tel =
 compileKeptTeleBind :: Hs.Name () -> Type -> C (Hs.TyVarBind ())
 compileKeptTeleBind x t = do
   checkValidTyVarName x
-  case compileKind t of
+  compileKind t >>= (\case
```

Comment:
Small nitpick: you don't need to have parentheses around the `\case ...` here. In general it's good style to avoid parentheses that span multiple lines.